dotnet_grpc_bench/GreeterServer/Program.cs

```diff
@@ -16,37 +16,38 @@
 
 #endregion
 
-using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+using GreeterServer.Services;
+using GreeterServer;
+using System;
 
-namespace GreeterServer
+var builder = WebApplication.CreateBuilder(args);
+
+builder.WebHost.ConfigureKestrel(options =>
 {
-    public class Program
-    {
-        public static void Main(string[] args)
-        {
-            CreateHostBuilder(args).Build().Run();
-        }
-
-        public static IHostBuilder CreateHostBuilder(string[] args) =>
-            Host.CreateDefaultBuilder(args)
-                .ConfigureWebHostDefaults(webBuilder =>
-                {
-                    webBuilder.ConfigureKestrel(options =>
-                    {
-                        options.AddServerHeader = false;
-                        options.ListenAnyIP(50051, listenOptions =>
-                        {
-                            listenOptions.Protocols = HttpProtocols.Http2;
-                        });
-                    });
-                    webBuilder.ConfigureLogging(logging =>
-                    {
-                        logging.ClearProviders();
-                    });
-                    webBuilder.UseStartup<Startup>();
-                });
-    }
-}
+	options.AddServerHeader = false;
+	options.ListenAnyIP(50051, listenOptions =>
+	{
+		listenOptions.Protocols = HttpProtocols.Http2;
+	});
+});
+
+builder.Logging.ClearProviders();
+
+builder.Services.AddGrpc(o => o.IgnoreUnknownServices = true);
+builder.Services.Configure<RouteOptions>(c => c.SuppressCheckForUnhandledSecurityMetadata = true);
+builder.Services.AddSingleton<GreeterService>();
+
+var app = builder.Build();
+
+app.Lifetime.ApplicationStarted.Register(() => Console.WriteLine("Application started."));
```

Comment:
It was there in the initial version as well. My guess is since there isn’t any logging we don’t really know if the service is running or not. This can give us an indication that its ready.

Reply:
This code was copied from the ASP.NET Core benchmarks which writes to the console to let the system know that it is ready. I guess it is not needed.

Reply:
Does this hurt perf? Isnt it better to have it?

Reply:
It's fine. I guess we turned off logging so this helps.